.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Check and build codebase
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [ 16.x, 18.x ]
```

Comment:
any particular reason we are running on 16 and 18 ? or just for funsies?

Reply:
I'm open to suggestions on this. Running on a matrix definitely seems more applicable to libraries, but I tend to run on LTS and the next stable version. I'm assuming Vercel is on Node 16 right now, but will eventually upgrade to 18 – so it's nice to know ahead of time what's going to break in future versions.

We could probably, realistically, just reduce this to Node 16. What do you think?

Reply:
since this isn't a client project, lets keep it on both. 

i would love to see how long it takes a PR to run through all its checks between vercel and our tests when modifying multiple apps.

Reply:
Yeah for sure, I'd also be interested in setting up the TurboRepo shared cache (or whatever) and see what that does to build times as well!